lib/src/code_field/code_field.dart

```diff
@@ -89,8 +90,8 @@ class CodeFieldState extends State<CodeField> {
   ScrollController? _horizontalCodeScroll;
   final _codeFieldKey = GlobalKey();
 
-  double cursorX = 0;
-  double cursorY = 0;
+  double popupLeftOffset = 0;
+  double popupTopOffset = 0;
```

Comment:
If this widget calculates the offset once and does not adjust it later, then this can also be a single `Offset preferredPopupOffset` object.